sway-core/src/semantic_analysis/symbol_resolve_context.rs

```diff
@@ -43,6 +43,8 @@ pub struct SymbolResolveContext<'a> {
     ///
     /// This is `Disallow` everywhere except while checking type parameters bounds in struct instantiation.
     generic_shadowing_mode: GenericShadowingMode,
+    /// Visibility checking level for the current scope.
+    visibility_level: Visibility,
```

Comment:
I don't understand this comment, and I'm finding it difficult to understand what `visibility_level` is supposed to represent. Could you elaborate a bit (ideally in this comment, or in the PR description)?